File: backend/internal/wishlist/wish.go
package wishlist

import "time"

// Wish — сущность из БД. Без json-тегов: наружу отдаём WishResponse.
type Wish struct {
	ID            int
	OwnerID       int
	OwnerUsername string
	Title         string
	Description   *string
	URL           *string
	Price         *float64
	Status        string
	GroupName     string
	GroupColor    string
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

File: frontend/src/types/wish.ts
export type Wish = {
  id: number;
  owner_id: number;
  owner_username: string;
  title: string;
  description: string | null;
  url: string | null;
  price: number | null;
  status: "wanted" | "reserved" | "done";
  group_name: string;
  group_color: GroupColor;
  created_at: string;
  updated_at: string;
};

export type GroupColor = "slate" | "blue" | "green" | "violet" | "rose";

export type CreateWishPayload = {
  title: string;
  description?: string;
  url?: string;
  price?: number;
  status?: "wanted" | "reserved" | "done";
  group_name?: string;
  group_color?: GroupColor;
};

export type UpdateWishPayload = {
  description?: string;
  url?: string;
  price?: number;
};
